src/app/pages/ProposalDetailsPage/hooks.ts

```diff
@@ -0,0 +1,179 @@
+  return {
+    isLoading,
+    isError,
+    map: (data?.data as ExtendedValidatorList)?.map ?? new Map<string, Validator>(),
+  }
+}
+
+export const useAllVotes = (network: Network, proposalId: number): AllVotesData => {
+  const query = useGetConsensusProposalsProposalIdVotes(network, proposalId)
+  const {
+    map: validators,
+    isLoading: areValidatorsLoading,
+    isError: haveValidatorsFailed,
+  } = useValidatorMap(network)
+  const { isLoading, isError, data } = query
+
+  const extendedVotes = (data?.data.votes || []).map(
+    (vote, index): ExtendedVote => ({
+      ...vote,
+      index,
+      areValidatorsLoading,
+      haveValidatorsFailed,
+      validator: validators.get(vote.address),
+    }),
+  )
+
+  return {
+    isLoading,
+    isError,
+    loadedVotes: DEBUG_MODE
+      ? extendedVotes.map(v => ({ ...v, vote: getRandomVoteFor(v.address) })) || []
+      : extendedVotes,
+    total_count: data?.data.total_count ?? 0, // We need fallbacks here so that we always satisfy the List interface
+    is_total_count_clipped: data?.data.is_total_count_clipped ?? false,
+  }
+}
+
+export type VoteStats = {
+  isLoading: boolean
+  isError: boolean
+
+  /**
+   * Did we manage to get all data from the server?
+   */
+  isComplete: boolean
+
+  /**
+   * The results of counting the votes
+   */
+  tally: Record<ProposalVoteValue, bigint>
+
+  /**
+   * The total number of (valid) votes
+   */
+  allVotesCount: number
+
+  /**
+   * The total amount of valid votes (considering the shares)
+   */
+  allVotesPower: bigint
+}
+
+export const useVoteStats = (network: Network, proposalId: number): VoteStats => {
+  const { isLoading, isError, loadedVotes, is_total_count_clipped } = useAllVotes(network, proposalId)
+  const tally: Record<ProposalVoteValue, bigint> = {
+    yes: 0n,
+    no: 0n,
+    abstain: 0n,
+  }
+  // TODO: instead of 1n, we should add the power of the vote, but that data is not available yet.
+  loadedVotes.forEach(vote => (tally[vote.vote as ProposalVoteValue] += 1n))
+  const allVotesCount = loadedVotes.length
+  const allVotesPower = tally.yes + tally.no + tally.abstain
+  const isComplete = !isError && !is_total_count_clipped
+  return {
+    isLoading,
+    isError,
+    isComplete,
+    tally,
+    allVotesCount,
+    allVotesPower,
+  }
+}
+
+export const useWantedVoteType = () =>
+  useTypedSearchParam<VoteType>('vote', 'any', {
+    deleteParams: ['page'],
+  })
+
+export const useVoterSearch = () => useTypedSearchParam('voter', '', { deleteParams: ['page'] })
+
+export const useVoterSearchPattern = () => {
+  const [voterSearchInput] = useVoterSearch()
+  return voterSearchInput.length < 3 ? undefined : voterSearchInput
+}
+
+export const useVoterSearchError = (t: TFunction) => {
+  const [input] = useVoterSearch()
+  const pattern = useVoterSearchPattern()
+  return !!input && !pattern ? t('tableSearch.error.tooShort') : undefined
+}
```

Comment:
This is bad design: developer basically can not know which of these 3 to use and what it does without reading the code of all three of these functions.

What would be your ideal interface / function shape for consuming this?

Reply:
> developer basically can not know which of these 3 to use and what it does without reading the code of all three of these functions.

True.

> What would be your ideal interface / function shape for consuming this?

That's a non-trivial question. 
- Usually I prefer returning objects as opposed to arrays, because with field names there is no confusion, and also it's easier to selectively pick the field I want. 
- However, because or `React.useState()`, the `[value, setValue()]` pattern is very widespread.

Given all, that,
 - For ease of use, I kind of want to keep using this pattern for the obvious things (value and setter),
 - For anything,else, something else. Either individual hooks, or a bigger hook that provides everything in a object.

But how to have both side by side, and not be confusing? 

(Use longer names that exactly describes the functionality? Have only a single hook, with everything there, and let go of the ease of use of `[value, setValue()]` ?)